compiler/ballerina-backend-jvm/src/main/ballerina/src/compiler_backend_jvm/jvm_terminator_gen.bal

```diff
@@ -688,7 +688,34 @@ type TerminatorGenerator object {
         lambdas[lambdaName] = callIns;
         lambdaIndex += 1;
 
-        self.submitToScheduler(callIns.lhsOp, localVarOffset);
+        boolean concurrent = false;
+        // check for concurrent annotation
+        if (callIns.annotAttachments.length() > 0 ) {
+            foreach var v in callIns.annotAttachments {                
+                if (v is bir:AnnotationAttachment && 
+                    v.annotTagRef.value == "strand" && 
+                    v.moduleId.org == "ballerina" && 
+                    v.moduleId.name == "lang.annotations") {
+                        if (v.annotValues.length() > 0) {
+                            bir:AnnotationValue val = <bir:AnnotationValue> v.annotValues[0];
+                            if (val is bir:AnnotationRecordValue) {
+                                if (val.annotValueMap.hasKey("thread") && 
+                                    val.annotValueMap.get("thread")["literalValue"] == "any") {
+                                        concurrent = true;
+                                }
+
+                                if (val.annotValueMap.hasKey("name") &&
+                                    val.annotValueMap.get("name")["literalValue"] != "DEFAULT") {
+                                        panic error("Unsupported policy. Only 'DEFAULT' policy is supported by jballerina runtime.");
+                                }
```

Comment:
why do we need this ? Can there be a place use would use "DEFAULT" ?

Reply:
This is for future proof. Not required right now. There can be other types in the future.